src/braket/circuits/circuit.py

```diff
@@ -196,6 +201,28 @@ def qubits(self) -> QubitSet:
         """QubitSet: Get a copy of the qubits for this circuit."""
         return QubitSet(self._moments.qubits.union(self._qubit_observable_set))
 
+    @property
+    def has_free_parameters(self) -> bool:
```

Comment:
I'm worried that we're going to have trouble tracking this. If this is just an optimization, how expensive would it be to instead check the circuit before compilation?

Reply:
Depends on the circuit size. This check is being used to note the existence of FreeParameters in the circuit. The plan for execution is to have the values replace at the time of execution. This will serve as a means of going through circuits that need this replacement.